lib/logstash/inputs/twitter.rb

```diff
@@ -85,31 +91,25 @@ def on_body(data)
   def run(queue)
     @logger.info("Starting twitter tracking", :keywords => @keywords)
     begin
-      @client.filter(:track => @keywords.join(",")) do |tweet|
-        if tweet.is_a?(Twitter::Tweet)
-          @logger.debug? && @logger.debug("Got tweet", :user => tweet.user.screen_name, :text => tweet.text)
-          if @full_tweet
-            event = LogStash::Event.new(LogStash::Util.stringify_symbols(tweet.to_hash))
-            event.timestamp = LogStash::Timestamp.new(tweet.created_at)
-          else
-            event = LogStash::Event.new(
-              LogStash::Event::TIMESTAMP => LogStash::Timestamp.new(tweet.created_at),
-              "message" => tweet.full_text,
-              "user" => tweet.user.screen_name,
-              "client" => tweet.source,
-              "retweeted" => tweet.retweeted?,
-              "source" => "http://twitter.com/#{tweet.user.screen_name}/status/#{tweet.id}"
-            )
-            event["in-reply-to"] = tweet.in_reply_to_status_id if tweet.reply?
-            unless tweet.urls.empty?
-              event["urls"] = tweet.urls.map(&:expanded_url).map(&:to_s)
-            end
-          end
+      options = {}
+      if defined? @keywords and @keywords.length > 0
+        options[:track] = @keywords.join(",")
+      end
+      if defined? @locations and @locations.length > 0
+        options[:locations] = @locations.join(",")
+      end
+      if defined? @follows and @follows.length > 0
+        options[:follow] = @follows.join(",")
+      end
+
+      if options.empty?
+        @logger.debug? && @logger.debug("Listening on the firehose")
+        @client.firehose() { |tweet|  process_a_tweet( tweet, queue)}
+      else
+        @logger.debug? && @logger.debug("Filtering for tweets", :keys => options.keys)
```

Comment:
it might be easier to debug if instead of logging the keys, you log the options hash: `:options => options`